x/poolmanager/README.md

```diff
@@ -330,3 +330,52 @@ higher slippage.
 
 Note, that the actual split happens off-chain. The router is only responsible for executing the swaps in the order and quantities of token in provided
 by the routes.
+
+## EstimateTradeBasedOnPriceImpact Query
+
+The `EstimateTradeBasedOnPriceImpact` query allows users to estimate a trade for all pool types given the following parameters are provided for this request `EstimateTradeBasedOnPriceImpactRequest`:
+
+
+- **FromCoin**: (`sdk.Coin`): the total amount of tokens one wants to sell.
+- **ToCoinDenom**: (`string`): is the denom they want to buy with the tokens being sold.
+- **PoolId**: (`uint64`): is the identifier of the pool that the trade will happen on.
+- **MaxPriceImpact**: (`sdk.Dec`): is the maximum percentage that the user is willing to affect the price of the pool.
+- **TwapPrice**: (`sdk.Dec`) is an external price that the user can optionally enter to have the `MaxPriceImpact` adjusted as the `SpotPrice` of a pool could be changed at any time.
+
+The response would be `EstimateTradeBasedOnPriceImpactResponse` which contains the following data:
+
+- **InputCoin**: (`sdk.Coin`): the actual input amount that would be tradeable under that price impact (might be the full amount).
+- **OutputCoin**: (`sdk.Coin`): the amount of the `ToCoinDenom` tokens being received for the actual `InputCoin` trade.
+
+With that data it is easier for any entity to fill in the `MsgSwapExactAmountIn` details.
+
+### Process
```

Comment:
To be updated in line with changes applied for: https://github.com/entrypoint-zone/osmosis/pull/1/files#r1297076185